mrchecker-framework-modules/mrchecker-selenium-module/src/main/java/com/capgemini/mrchecker/selenium/core/newDrivers/DriverManager.java

```diff
@@ -277,41 +294,44 @@ public INewWebDriver getDriver() {
 			public INewWebDriver getDriver() {
 				String browserPath = DriverManager.propertiesSelenium.getSeleniumIE();
 				boolean isDriverAutoUpdateActivated = DriverManager.propertiesSelenium.getDriverAutoUpdateFlag();
-				
-				if (isDriverAutoUpdateActivated) {
-					downloadNewestVersionOfWebDriver(InternetExplorerDriver.class);
-					OperationsOnFiles.moveWithPruneEmptydirectories(WebDriverManager.getInstance(InternetExplorerDriver.class)
-							.getBinaryPath(), browserPath);
+				synchronized (this) {
```

Comment:
Those `synchronized` blocks could be extracted to the method to avoid code duplication

Reply:
@tompalka 
If I put synchronized blocks in function then it will change the behavior, because now every driver can be downloaded at the same time and after change only one driver at one time. Do we want such change? 
I also work with existing code. Somebody in the past implemented that in such way without function. My changes are not big. I thought about refactoring that, but there will be a lot of ifs in such function because in such function we would work with many variables connected to specified driver. 
Can we leave it that way?

Reply:
What about extracting just this part:
`
downloadNewestOrGivenVersionOfWebDriver(FirefoxDriver.class);
						OperationsOnFiles.moveWithPruneEmptydirectories(WebDriverManager.getInstance(FirefoxDriver.class)
								.getBinaryPath(), browserPath);
`
passing just proper class to the method? It will help further refactor those duplications.

Reply:
> What about extracting just this part:
> `downloadNewestOrGivenVersionOfWebDriver(FirefoxDriver.class); OperationsOnFiles.moveWithPruneEmptydirectories(WebDriverManager.getInstance(FirefoxDriver.class) .getBinaryPath(), browserPath);`
> passing just proper class to the method? It will help further refactor those duplications.

Discussed, issue created: https://github.com/devonfw/devonfw-testing/issues/196